pkg/mtbroker/filter/filter_handler.go

```diff
@@ -102,7 +102,7 @@ func NewHandler(logger *zap.Logger, triggerLister eventinglisters.TriggerLister,
 //
 // This method will block until ctx is done.
 func (h *Handler) Start(ctx context.Context) error {
-	return h.receiver.StartListen(ctx, health.WithLivenessCheck(health.WithReadinessCheck(h)))
+	return h.receiver.StartListen(ctx, &handlers.Drainer{Inner: h})
```

Comment:
I think that you need something that calls `Drain()` when `ctx.Done()` is closed, see: https://github.com/knative/pkg/blob/673521e55e5b9f081b672c2f97c81c31b1cd2ddc/webhook/webhook.go#L218-L228

Reply:
ping @vaikas